types/web3/web3-tests.ts

```diff
@@ -39,6 +39,38 @@ myContract.options.gas = 5000000;
 //
 // web3.eth.accounts
 // --------------------------------------------------------------------------
+const account = web3.eth.accounts.privateKeyToAccount("0x1234");
+
+// check that no `publicKey` field is present on `Account` type
+const noPublicKeyInAccount: typeof account & { publicKey?: never } = account;
```

Comment:
Is this check not kinda random? There is an infinite number of other fields that also should not be present on the Account type. I do not think that we should introduce such a "blacklist check". I would remove this and all other checks of the same type below. It is technically correct but confusing to read and there is no "logic" behind the question of when or when not to add such a check elsewhere in the project.

I thus will suggest changes. If you want @zlumer I will approve anyhow.

Reply:
I agree that there's an infinite number of excluded fields and the test looks a little dirty.
This check was meant to add an additional layer of regression protection, since the field existed (incorrectly) before. I'm not sure if this goal was achieved, but I was trying to handle the case where someone updates `web3` type definitions on a (unknowingly buggy) live project, sees compiler errors and makes a PR returning these deleted fields.

This PR was already merged, but if you insist we should remove the checks, I can make another PR.